tools/hsmtool.c

```diff
@@ -67,17 +70,25 @@ static bool ensure_hsm_secret_exists(int fd, const char *path)
 	return true;
 }
 
-static void get_hsm_secret(struct secret *hsm_secret,
-                           const char *hsm_secret_path)
+static void grab_hsm_file(const char *hsm_secret_path,
+			  void *dst, size_t dstlen)
 {
-	int fd;
+	u8 *contents = grab_file(tmpctx, hsm_secret_path);
+	if (!contents)
+		errx(EXITCODE_ERROR_HSM_FILE, "Reading hsm_secret");
 
-	fd = open(hsm_secret_path, O_RDONLY);
-	if (fd < 0)
-		errx(EXITCODE_ERROR_HSM_FILE, "Could not open hsm_secret");
-	if (!read_all(fd, hsm_secret, sizeof(*hsm_secret)))
-		errx(EXITCODE_ERROR_HSM_FILE, "Could not read hsm_secret");
-	close(fd);
+	/* grab_file always appends a NUL char for convenience */
```

Comment:
```suggestion
	/* grab_file always appends a NULL char for convenience */
```

Reply:
No, confusingly "NUL" is the ASCII zero, NULL is the C pointer!